drivers/metadata/informationschema/metadata.go

```diff
@@ -821,6 +851,179 @@ func (s InformationSchema) Sequences(f metadata.Filter) (*metadata.SequenceSet,
+	if !s.hasTablePrivileges && !s.hasColumnPrivileges && !s.hasUsagePrivileges {
+		return nil, text.ErrNotSupported
+	}
+
+	qstrs := []string{}
+	conds, vals := s.conditions(1, f, formats{
+		catalog:    "object_catalog LIKE %s",
+		schema:     "object_schema LIKE %s",
+		notSchemas: "object_schema NOT IN (%s)",
+		name:       "object_name LIKE %s",
+		types:      "object_type IN (%s)",
+	})
+
+	if s.hasTablePrivileges {
+		columns := []string{
+			"t.table_catalog AS object_catalog",
+			"t.table_schema AS object_schema",
+			"t.table_name AS object_name",
+			"t.table_type AS object_type",
+			"'' AS column_name",
+			"COALESCE(grantee, '') AS grantee",
+			"COALESCE(" + s.clauses[PrivilegesGrantor] + ", '') AS grantor",
+			"COALESCE(privilege_type, '') AS privilege_type",
+			"CASE WHEN is_grantable='YES' THEN 1 ELSE 0 END AS is_grantable",
+		}
+
+		// `tables` is on the left side of the join to also list tables that have no privileges set.
+		qstr := "SELECT\n" +
+			"  " + strings.Join(columns, ", ") + "\n" +
+			"FROM information_schema.tables t\n" +
+			"LEFT JOIN information_schema.table_privileges tp\n" +
+			"  ON t.table_catalog = tp.table_catalog AND t.table_schema = tp.table_schema AND t.table_name = tp.table_name"
+		qstrs = append(qstrs, qstr)
+	}
+
+	if s.hasColumnPrivileges {
+		columns := []string{
+			"t.table_catalog AS object_catalog",
+			"t.table_schema AS object_schema",
+			"t.table_name AS object_name",
+			"t.table_type AS object_type",
+			"column_name",
+			"grantee",
+			s.clauses[PrivilegesGrantor] + " AS grantor",
+			"privilege_type",
+			"CASE WHEN is_grantable='YES' THEN 1 ELSE 0 END AS is_grantable",
+		}
+
+		qstr := "SELECT\n" +
+			"  " + strings.Join(columns, ", ") + "\n" +
+			"FROM information_schema.column_privileges cp\n" +
+			"LEFT JOIN information_schema.tables t\n" +
+			"  ON t.table_catalog = cp.table_catalog AND t.table_schema = cp.table_schema AND t.table_name = cp.table_name"
+		qstrs = append(qstrs, qstr)
+	}
+
+	if s.hasUsagePrivileges {
+		columns := []string{
+			"object_catalog",
+			"object_schema",
+			"object_name",
+			"object_type",
+			"'' AS column_name",
+			"grantee",
+			s.clauses[PrivilegesGrantor] + " AS grantor",
+			"privilege_type",
+			"CASE WHEN is_grantable='YES' THEN 1 ELSE 0 END AS is_grantable",
+		}
+
+		qstr := "SELECT\n" +
+			"  " + strings.Join(columns, ", ") + "\n" +
+			"FROM information_schema.usage_privileges"
+		qstrs = append(qstrs, qstr)
+	}
+
+	// In the query result, table and column level privileges will be on separate rows.
+	// Each table or column can have multple privileges (i.e rows).
+	// For table level privileges the `column_name` column is empty.
+	qstr := "SELECT * FROM (\n" + strings.Join(qstrs, "\nUNION ALL\n") + "\n) AS subquery"
+	rows, closeRows, err := s.query(
+		qstr,
+		conds,
+		"object_catalog, object_schema, object_name, object_type, column_name, grantee, grantor, privilege_type",
```

Comment:
Should results be sorted by object type first, and then object name?

Reply:
To me that would make sense.

The information schema implementation for \d sorts by object type too. But psql and our postgres implementation of \d sort by object name. I would propose to change the \d postgres implementation to object type as well to make it consistent.

Reply:
I made the changes for both the postgres reader and the information schema reader.